src/evaluator.ts

```diff
@@ -536,7 +539,11 @@ class Evaluator {
     const assertionTypes = new Set<string>();
     const rowsWithSelectBestAssertion = new Set<number>();
 
-    await runExtensionHook(testSuite.extensions, 'beforeAll', { suite: testSuite });
+    const beforeAllExtensionOutputs = hasDefinedExtensionHooks(testSuite.extensions)
+      ? await runExtensionHook(testSuite.extensions!, 'beforeAll', {
+          suite: testSuite,
+        })
+      : [];
```

Comment:
is empty array the right fallback here? I would've expected `undefined`

Reply:
That's a good point. I set it up to be consistent with `extensionHookOutputs` while avoiding having undefined properties on `extensionHookOutputs` (because it gets confusing whether a property was set as `undefined` or never set in the first place).  `null` might be a suitable alternative but it evals the same as the empty array.